components/server/src/github/github-context-parser.ts

```diff
@@ -302,6 +302,9 @@ export class GithubContextParser extends AbstractContextParser implements IConte
                     throw new Error(`Couldn't find issue or pull request #${pullRequestNr} in repository ${owner}/${repoName}.`)
                 }
             }
+            if (pr.headRef === null) {
+                throw new Error(`Couldn't find branch for pull request #${pullRequestNr}. Maybe the branch (or fork) has been deleted?`);
```

Comment:
Good points, thanks!

> **question**: Do we need to mention the fork here? Feel free to adjust or leave this as is. Your call! 📞

We don't necessarily have to (although one issue specifically mentioned deleted forks: https://github.com/gitpod-io/gitpod/issues/1375), because technically deleting a fork also deletes the source branch of the PR.

I'll remove it to simplify the message! 🙌

> 1. Regarding voice and tone we could try avoiding abbreviations like _couldn't_ to make the message more clear.

Makes sense, I agree.

> Could not open pull request #${pullRequestNr}. Source branch or fork project have been removed.

- As mentioned, I'll drop the "fork".

- If I'm pessimistic or confused, this might read like "Gitpod got confused about this PR, and has deleted your code." -- I'll try to find a formulation that cannot possibly be interpreted as "Gitpod has actively deleted something" 😅